libs/cases/frontend/view/src/lib/components/detail-case/detail-case.component.ts

```diff
@@ -27,5 +30,135 @@ export class DetailCaseComponent {
+            this.onSubmit();
+        } else {
+            this.isEditing = true;
+        }
+    }
+
+    onSubmit(): void {
+        // TODO validate form
+        if (this.validateForm() === 'valid') {
+            const caseId = this.mapCaseId(this.casedetail());
+            const caseData = this.casedetail();
+
+            if (caseId !== undefined && caseData !== undefined) {
+                this._browseFacade.updateCase(caseId, caseData).subscribe({});
+            }
+            this.isEditing = false;
+        } else {
+            this.showErrorToast(this.validateForm());
+        }
+    }
+
+    validateForm() {
+        const casedetail = this.casedetail();
+
+        if (casedetail !== undefined) {
+
+            if (this.casedetail()?.title === '') {
+                return 'Empty title';
+            }
+
+
+            if (casedetail.status !== undefined) {
+                if (!Object.values(ECaseStatus).includes(casedetail.status)) {
+                    return 'Status is not in List: OPEN, INPROGRESS, OVERDUE, ONHOLD, DONE, CANCELLED, ARCHIVED';
+                }
+            }
+            if (casedetail.priority !== undefined) {
+            if (!Object.values(ECasePriority).includes(casedetail.priority)) {
+                return 'Priority is not in List: EMERGENCY, HIGH, MEDIUM, LOW';
+            }}
+            if (!(this.casedetail()?.modifiedBy.includes('@') && this.casedetail()?.modifiedBy.includes('.'))) {
+                return 'Invalid email';
+            }
+            if (casedetail.type !== undefined) {
+            if (!Object.values(ECaseType).includes(casedetail.type)) {
+                return 'Priority is not in List: PLANNED, INNCIDENT, ANNOTATION';
+            }}
+            if (casedetail.dueDate !== undefined) {
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-expect-error
+            if (!this.datePattern.test(casedetail.dueDate)) {
+                return 'Invalid date format';
+            }}
+            const dueDate = this.casedetail()?.dueDate;
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-expect-error
+            if (dueDate && this.datePattern.test(this.casedetail()?.dueDate)) {
+                const match = this.casedetail()?.dueDate;
+                if (match) {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-expect-error
+                    const month: int = (parseInt(match[5])) * 10 + parseInt(match[6]);
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-expect-error
+                    const day = parseInt(match[8]) * 10 + parseInt(match[9]);
+                    if (!(month >= 1 && month <= 12 && day >= 1 && day <= 31)) {
+                        return 'Invalid month or day in due date';
+                    }
+                }
```

Comment:
the date checking is still not that good, I will try to see if I can find a better way